### PR TITLE
UCP/API: Add tag_sender_mask to params

### PR DESCRIPTION
**Pull request by brminich**

### CODE REVIEW

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -328,6 +328,12 @@ typedef struct ucp_params {
      */
     ucp_request_cleanup_callback_t     request_cleanup;
     /**
+     * Mask which specifies particular bits of the tag which can uniquely
+     * identify the sender in tagged operations.
+     * This field defaults to 0 if not specified.
+     */
+    uint64_t                           tag_sender_mask;
```
Comment (yosefe) [inline src/ucp/api/ucp.h]: need to add a bit for it in field_mask

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2322/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1647/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2323/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1648/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
:+1: 
@shamisp pls review as well

File: src/ucp/api/ucp.h (hunk 1 above)
Comment (shamisp) [inline src/ucp/api/ucp.h]: ".. identify the sender ... " - a.k.a SRC rank (or PE index) ?

Comment (yosefe) [inline src/ucp/api/ucp.h]: in API terminology, this is "ucp_ep"

Comment (shamisp) [inline src/ucp/api/ucp.h]: Then I would suggest minor change:
s/sender/sender (UCP endpoint)/

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2325/ for details.

**Comment by shamisp** [comment]
👍

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1650/ for details (Mellanox internal link).

**Pull request merged**